<?xml version="1.0"?>

<launch>
  <arg name="mav_name" default="inspire1" />
  <arg name="namespace" default="$(arg mav_name)"/>
  <!-- <include file="$(find dji_sdk)/launch/dji_sdk_node.launch.xml">
    <arg name="serial_name" value="/dev/ttyUSB0"/>
    <arg name="drone_namespace" default="$(arg mav_name)" />
  </include> -->

  <node pkg="geodetic_utils" type="geodetic_to_local_conversion_node" name="geodetic_to_local_conversion_node" output="screen">
    <remap from="gps" to="/dji_sdk/gps_position"/>
    <remap from="imu" to="/dji_sdk/imu"/>
    <remap from="vel" to="/dji_sdk/velocity"/>
    <remap from="external_altitude" to="/dji_sdk/height_above_takeoff"/>
    <remap from="odometry" to="/dji_sdk/odometry"/>
  </node>


  <node pkg="geodetic_utils" type="gimbal_tf_broadcaster.py" name="gimbal_tf_broadcaster" respawn="true">
  </node>

  <group ns="$(arg namespace)">
    <rosparam command="load" file="$(find aerography)/param/aerography.yaml" />

    <!-- <node pkg="dji_command" type="rpy_convert_node.py" name="rpy_convert_node" respawn="true">
    </node> -->


    <!--node pkg="aerography" type="aerography" name="aerography" respawn="false"/-->

    <!-- <node pkg="mav_trajectory_generation_ros" type="trajectory_sampler_node" name="trajectory_sampler_node" respawn="true">
      <param name="publish_whole_trajectory" type="bool" value="true" />
    </node> -->

    <!--node pkg="tf" type="static_transform_publisher" name="velodyne" args="0 0 0 0 0 0 0 gps_receiver 100" /-->

    <node name="mav_nonlinear_mpc" pkg="mav_nonlinear_mpc" type="nonlinear_mpc_node" respawn="true" clear_params="true" output="screen">
      <remap from="odometry" to="/dji_sdk/odometry" />
      <rosparam file="$(find dji_command)/resources/nonlinear_mpc_$(arg mav_name).yaml" />
      <rosparam file="$(find dji_command)/resources/disturbance_observer_$(arg mav_name).yaml"/>
      <param name="use_rc_teleop" value="false"/>
      <param name="verbose" value="true" />
      <param name="reference_frame" value="world"/>
    </node>


    <node pkg="dji_command" type="set_trajectory_node" name="DJIAttitudeControllerNode" respawn="false"></node>

  </group>

</launch>
